tools/sync_tron_state_from_k8s.py

```diff
@@ -0,0 +1,220 @@
+
+    # We also don't want kube_client debug logs
+    kube_logger = logging.getLogger("kubernetes.client.rest")
+    kube_logger.setLevel(logging.INFO)
+
+    return args
+
+
+def fetch_pods(kubeconfig_path: str) -> Dict[str, V1Pod]:
+    kube_client = KubeClient(kubeconfig_path=kubeconfig_path, user_agent="sync_tron_state_from_k8s")
+
+    # Bit of a hack, no helper to fetch pods so reach into core api
+    completed_pod_list = kube_client.core.list_namespaced_pod(
+        namespace="tron",
+    )
+
+    return {pod.metadata.name: pod for pod in completed_pod_list.items}
+
+
+def get_tron_state_from_api(tron_server: str, num_runs: int = 100) -> List[Dict[str, Dict[Any, Any]]]:
+    if not tron_server:
+        client_config = get_client_config()
+        tron_server = client_config.get("server", "http://localhost:8089")
+    client = Client(tron_server)
+    # /jobs returns only the latest 5 runs, we'll need to request all runs instead ourselves
+    jobs = client.jobs(
+        include_job_runs=False,
+        include_action_runs=False,
+        include_action_graph=False,
+        include_node_pool=False,
+    )
+
+    for job in jobs:
+        # Update job URL to be used with API instead of web
+        url = f'/api{job["url"]}'
+        log.debug(f'Fetching job {job["name"]} at {url}')
+        job_runs = client.job(
+            url,
+            include_action_runs=True,
+            count=num_runs,  # TODO: fetch job run_limit and use that for count ?
+        )
+        job["runs"] = job_runs["runs"]
+    return jobs
+
+
+def get_matching_pod(action_run: Dict[str, Any], pods: Dict[str, V1Pod]) -> Optional[V1Pod]:
+    """Given a tron action_run, try to find the right pod that matches."""
+    action_name = action_run["action_name"]
+    job_name = action_run["job_name"]
+    run_num = action_run["run_num"]
+    service, job = job_name.split(".")
+    instance_name = f"{job}.{action_name}"
+    sanitized_instance_name = limit_size_with_hash(instance_name)
+    matching_pods = sorted(
+        [
+            pod
+            for pod in pods.values()
+            if pod.metadata.labels["paasta.yelp.com/service"] == service
+            and pod.metadata.labels["paasta.yelp.com/instance"] == sanitized_instance_name
+            and pod.metadata.labels["tron.yelp.com/run_num"] == run_num
+        ],
+        # If action has retries, there will be multiple pods w/ same job_run; we only want the latest
+        key=lambda pod: pod.metadata.creation_timestamp,
+        reverse=True,
+    )
+    return (
+        matching_pods[0] if matching_pods and matching_pods[0].status.phase in POD_STATUS_TO_TRON_STATE.keys() else None
+    )
+
+
+def get_desired_state_from_pod(pod: V1Pod) -> str:
+    k8s_state = pod.status.phase
+    return POD_STATUS_TO_TRON_STATE.get(k8s_state, "NoMatch")
+
+
+def update_tron_from_pods(
+    jobs: List[Dict[str, Any]], pods: Dict[str, V1Pod], tronctl_wrapper: str = "tronctl", do_work: bool = True
```

Comment:
```suggestion
    jobs: List[Dict[str, Any]], pods: Dict[str, V1Pod], tronctl_wrapper: str = "tronctl", do_work: bool = False
```

would be nice just in case (and would match the script default)

Reply:
oops, no idea why I would default to true there, thanks for catching this!